src/symbols.c

```diff
@@ -1597,13 +1597,21 @@ static TMTag *find_best_goto_tag(GeanyDocument *doc, GPtrArray *tags)
 
 static GPtrArray *filter_tags(GPtrArray *tags, TMTag *current_tag, gboolean definition)
 {
+	GeanyDocument *doc = document_get_current();
+	guint current_line = sci_get_current_line(doc->editor->sci) + 1;
 	const TMTagType forward_types = tm_tag_prototype_t | tm_tag_externvar_t;
 	TMTag *tmtag, *last_tag = NULL;
+	const gchar *current_scope = NULL;
 	GPtrArray *filtered_tags = g_ptr_array_new();
 	guint i;
 
+	symbols_get_current_function(doc, &current_scope);
+
 	foreach_ptr_array(tmtag, i, tags)
 	{
+		if (!tm_workspace_is_autocomplete_tag(tmtag, doc->tm_file, current_line, current_scope))
```

Comment:
This is what you suggested doing here instead of having the condition duplicated: https://github.com/geany/geany/pull/3185#discussion_r862393644

Reply:
Right, I suggested to refactor the added if clause and you did a good job there. But now I'm puzzled why this new condition is relevant at all at this place.

Reply:
This is because for tag definition/declaration goto basically the same conditions like for autocompletion apply. Without this patch, if you perform the goto for e.g. variable `i`, you'd see all the local variables `i`'s in your project, which is a lot and irrelevant to the current function.

Maybe `tm_workspace_is_autocomplete_tag()` could be named a different way, I just couldn't come up with a better name.

Reply:
So "goto tag" is now constrained to the same tags as scope completion, because with all the local tags of other functions that would be too much? Can you just add this as a small comment please so this call doesn't look out-of-place?

Reply:
Done.